app/services/katello/applicability/applicable_content_helper.rb

```diff
@@ -0,0 +1,88 @@
+module Katello
+  module Applicability
+    class ApplicableContentHelper
+      attr_accessor :content_facet, :content_unit_class
+
+      def initialize(content_unit_class, content_facet = nil)
+        self.content_facet = content_facet
+        self.content_unit_class = content_unit_class
+      end
+
+      def calculate_and_import
+        to_add, to_remove = applicable_differences
+        ActiveRecord::Base.transaction do
+          insert(to_add) unless to_add.blank?
+          remove(to_remove) unless to_remove.blank?
+        end
+
+        # FIXME: Debug return
+        return [to_add, to_remove]
+      end
+
+      def content_ids
+        if self.content_unit_class == ::Katello::Erratum
+          # Query for all Errata ids that are attached to the host's applicable packages
+          return ::Katello::Erratum.joins("INNER JOIN katello_repository_errata ON katello_errata.id = \
+                                          katello_repository_errata.erratum_id").
+                                          where(id: ::Katello::ErratumPackage.
+                                                where(nvrea: self.content_facet.applicable_rpms.pluck(:nvra)).
+                                                pluck(:erratum_id)).pluck(:id).uniq
+        elsif self.content_unit_class == ::Katello::ModuleStream
+          fail NotImplementedError
+        else
+          # Query for applicable RPM ids
+          return ::Katello::Rpm.joins("INNER JOIN katello_repository_rpms ON \
+                                      katello_rpms.id = katello_repository_rpms.rpm_id").
+                                      joins("INNER JOIN katello_installed_packages ON \
+                                            katello_rpms.name = katello_installed_packages.name AND \
+                                            katello_rpms.arch = katello_installed_packages.arch AND \
+                                            katello_rpms.evr > katello_installed_packages.evr").
+                                            joins("INNER JOIN katello_host_installed_packages ON \
+                                                  katello_installed_packages.id = \
+                                                  katello_host_installed_packages.installed_package_id WHERE \
+                                                  katello_repository_rpms.repository_id in \
+                                                  (#{self.content_facet.bound_repositories.pluck(:id).join(',')}) \
```

Comment:
it looks like you're using the 'bound' repositories, and not the library instances?

Reply:
Are you talking about the `bound_repository`s' library instances?  If I check my host's `bound_repositories`, all of their `library_instance`s return `nil`.

Reply:
They are nil because they are already library instances.  I need to grab the library instance if `library_instance` isn't nil.